library/Core.cpp

```diff
@@ -2769,6 +2769,37 @@ std::string Core::GetAliasCommand(const std::string &name, const std::string &de
         return default_;
 }
 
+Core::LowerScreen::LowerScreen(df::viewscreen* screen) :
+    screen_{screen}
+{
+    if (screen_->parent == &df::global::gview->view) {
+        screen_ = nullptr;
+        return;
+    }
+    swap(screen_->parent, screen_);
+}
+Core::LowerScreen::~LowerScreen()
+{
+    if (screen_)
+        swap(screen_, screen_->child);
+}
+
+void Core::LowerScreen::swap(df::viewscreen* a, df::viewscreen* b)
+{
+    df::viewscreen* ap = a->parent;
+    df::viewscreen* bc = b->child;
+
+    ap->child = b;
+    b->parent = ap;
+
+    a->parent = b;
+    b->child = a;
+
+    if (bc) bc->parent = a;
+    else Core::getInstance().top_viewscreen = a;
```

Comment:
Does anything important actually use this? I was under the impression that things like spotclean used getCurViewscreen(), which doesn't use this.

Reply:
Gui::cursor:hotkey() gets it from Plugin::invoke() which uses Core::getTopViewscreen()